packages/meditrak-app/app/assessment/specificQuestions/UserQuestion.jsx

```diff
@@ -0,0 +1,117 @@
+/**
+ * Tupaia
+ * Copyright (c) 2017 - 2024 Beyond Essential Systems Pty Ltd
+ */
+
+import React, { useState } from 'react';
+import { View } from 'react-native';
+import PropTypes from 'prop-types';
+import { connect } from 'react-redux';
+import { takeScrollControl, releaseScrollControl } from '../actions';
+import { getQuestion } from '../selectors';
+import { Autocomplete } from '../../widgets/Autocomplete/Autocomplete';
+
+const OPTIONS_PER_PAGE = 10;
+
+const UserQuestionComponent = props => {
+  const { selectedUserId, users, onSelectUser, scrollIntoFocus } = props;
+  const [searchTerm, setSearchTerm] = useState('');
+  const [maxResults, setMaxResults] = useState(OPTIONS_PER_PAGE);
+
+  const userList = users?.map(user => user.name) ?? [];
+
+  const generateOptionList = () => {
+    if (!searchTerm) {
+      return userList.slice(0, maxResults);
+    }
+    const lowercaseSearchTerm = searchTerm.toLowerCase();
+    const usersThatStartWithSearchTerm = userList.filter(user =>
+      user.toLowerCase().startsWith(lowercaseSearchTerm),
+    );
+
+    const usersThatContainSearchTerm = userList.filter(user =>
+      user.toLowerCase().includes(lowercaseSearchTerm),
+    );
+
+    // return first the users that start with the search term, then the users that contain the search term
+    return [...usersThatStartWithSearchTerm, ...usersThatContainSearchTerm].slice(0, maxResults);
+  };
+
+  const optionList = generateOptionList();
+
+  const getSelectedUser = () => users.find(user => user.id === selectedUserId);
+
+  const selectedUser = getSelectedUser();
+
+  const handleEndReached = () => {
+    // If we've reached the end of the list, don't try to load more
+    if (users.length <= OPTIONS_PER_PAGE || maxResults >= users.length) return;
+
+    const newMaxResults = maxResults + OPTIONS_PER_PAGE;
+
+    setMaxResults(newMaxResults);
+  };
+
+  const handleSelectOption = option => {
+    const newSelectedUser = users.find(user => user.name === option);
+    onSelectUser(newSelectedUser?.id ?? null);
```

Comment:
```suggestion
    if (!newSelectedUser) {
      throw new Error(`Cannot find user in database: ${option}`);
    }
    onSelectUser(newSelectedUser.id);
```
This feels like it would never fail, but on the chance it does it feels like it'd be better to fail explicitly than silently :+1: 

Btw I guess we just don't handle multiple users with the same name huh? Guess that's fine but might be worth mentioning to Georgia

Reply:
Looking in the database, it does seem like we have quite a number of duplicate names:
```
select first_name || ' ' || last_name, count(*) from user_account ua group by first_name || ' ' || last_name order by count desc
```

I'm fairly certain almost all of these would be the same person just with multiple emails. However, I worry that things might not work for those users as there might be multiple entries in the dropdown for them and they'd only be able to select one...

Maybe lets chat with Georgia and workout if this is a data cleanup job? But we might also need to sync down emails and show them if the user has multiple accounts?

Reply:
Oh yeah, good point. I think there is a task with the Maui team at the moment to cleanup test users in the db

Reply:
I think this probably will still work - it will save the `userId` but this is just checking for the value to display as 'selected', but won't actually impact the saved value. Duplicate names is always going to be an issue, unfortunately, since there could be multiple 'John Smiths', but unless we show the email address, I'm not sure what we can do to fix that